crates/notion-core/src/manifest/mod.rs

```diff
@@ -92,27 +91,33 @@ impl Manifest {
         toolchain: serial::ToolchainSpec,
         package_file: PathBuf,
     ) -> Fallible<()> {
-        // parse the entire package.json file into a Value
-        let file = File::open(&package_file).unknown()?;
-        let mut v: serde_json::Value = serde_json::from_reader(file).unknown()?;
+        let get_file = || package_file.to_string_lossy().to_string();
 
-        // detect indentation in package.json
-        let mut contents = String::new();
-        let mut indent_file = File::open(&package_file).unknown()?;
-        indent_file.read_to_string(&mut contents).unknown()?;
-        let indent = detect_indent::detect_indent(&contents);
+        // parse the entire package.json file into a Value
+        let file = File::open(&package_file)
```

Comment:
We can simplify this flow *and* reduce the number of I/O operations and allocations all in a go, because we're *basically* always going to have to allocate this string in any successful case, and in the error case, serde would have been doing similar under the hood anyway:

```diff
-       let file = File::open(&package_file)
+       let contents = read_to_string(&package_file)
            .with_context(|_| ErrorDetails::PackageReadError { file: get_file() })?;
-       let mut v: serde_json::Value = serde_json::from_reader(file)
+       let mut v: serde_json::Value = serde_json::from_str(&contents)
            .with_context(|_| ErrorDetails::PackageParseError { file: get_file() })?;

        if let Some(map) = v.as_object_mut() {
            // detect indentation in package.json
-           let contents = read_to_string(&package_file)
-               .with_context(|_| ErrorDetails::PackageReadError { file: get_file() })?;
            let indent = detect_indent::detect_indent(&contents);
```

Reply:
Good catch, that is a lot simpler